# NCtool



Descrizione bella lunga di cosa ci sta qui, but before you need to dowload and install [NCboost](https://github.com/RausellLab/NCBoost)

or you can dowload ncboost directly from  [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.2537088.svg)](https://doi.org/10.5281/zenodo.2537088)  

## Elenco nuove funzioni:

```
./NCBoost_scripts/ncboost_annotate_INDELS.sh /pathto/inF.tsv /pathto/outF.tsv \
                                             /pathto/reference_genome.fasta
```
Descrizione veloce 

```
./NCBoost_scripts/ncboost_annotate_PCHIC.sh /pathto/inF.tsv /pathto/outF.tsv \
                                            /PCHIC_data.tx 
                                            5
```
Descrizione veloce 

```
./NCBoost_scripts/ncboost_viual.sh 
```
Descrizione veloce 

```
./NCBoost_scripts/ncboost_merge.sh
```
Descrizione veloce 


## Come installarlo

Get into the NCBoost folder with `cd /NCBoost` and run:

```
git clone https://github.com/apollo994/NC_scripts.git
cd /NC_scripts
```

```
./NC_script/pimp_my_NCBoost.sh
```

Descrizione veloce 


## Test



## References

1: Wang and Hakonarson; (2010). ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data. Nucleic Acids Res. 38, e164-e164.

2: di Iulio et al. (2018). The human noncoding genome defined by genetic diversity. Nat. Genet. 50, 333-337.



## Contact
Please address comments and questions about NCBoost to me

## License

Prendi e porta casa
